docs/articles/documentation/reference/test-api/testcontroller/dispatchEvent.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: docs
+title: t.dispatchEvent Method
+permalink: /documentation/reference/test-api/testcontroller/dispatchevent.html
+---
+# t.dispatchEvent Method
+
+```text
+t.dispatchEvent(target, eventName[, options]) → this | Promise<any>
+```
+
+> Important! The `t.dispatchEvent` method is not supported in Internet Explorer 11 and earlier.
+
+Fires a DOM event on the `target` element.
+
+Parameter              | Type                                              | Description
+---------------------- | ------------------------------------------------- | ----------------
+`target`               | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the event target. See [Select Target Elements](#select-target-elements).
+`eventName`            | String                                            | Event name. See [MDN Event Reference](https://developer.mozilla.org/en-US/docs/Web/Events)
+`options`&#160;*(optional)* | Object                                       | A set of action parameters. See [Options](#options).
+
+## Options
+
+Use the `options` object to specify parameters for events fired with `t.dispatchEvent`.
+
+All DOM events support the following `options` properties:
+
+* [Event Interface Properties](https://developer.mozilla.org/en-US/docs/Web/API/Event#properties)
+
+Unless otherwise specified, events have `bubbles` and `cancelable` options set to `true`.
+
+Depending on the `eventName` parameter, a different constructor is called for an event and different sets of additional `options` are available.
+
+You can explicitly set a constructor for an event. See [Explicit Constructor Assignment](#explicit-constructor-assignment).
+
+### Mouse Events
+
+You can use `t.dispatchEvent` to trigger a mouse event (like [mousedown](https://developer.mozilla.org/en-US/docs/Web/API/Element/mousedown_event) or [dblclick](https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event)).
+
+An example below fires the [mousedown](https://developer.mozilla.org/en-US/docs/Web/API/Element/mousedown_event) and [mouseup](https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseup_event) events to simulate the user holding down a left click on a button for 5 seconds.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture`Trigger Mouse Events`
+    .page('www.example.com');
+
+test('Left-click a button for 5 seconds', async t => {
+    const target = Selector('#target');
+
+    await t
+        .dispatchEvent(target, 'mousedown')
+        .wait(5000)
+        .dispatchEvent(target, 'mouseup');
+});
+```
+
+Mouse events support the following `options` properties:
+
+* [Event Properties](https://developer.mozilla.org/en-US/docs/Web/API/Event#properties)
+* [UIEvent Properties](https://developer.mozilla.org/en-US/docs/Web/API/UIEvent#properties)
+* [MouseEvent Properties](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent#properties)
```

Comment:
These links are not useful.

You can just specify the all options that are can be passed to the `MouseEvent` class constructor.
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/MouseEvent